### PR TITLE
Training on Synthea Medications dataset through Tensorflow

### LINKED ISSUE #70: Train Model on Synthea data
## Feature Description
Train a machine learning model (not a split network) on synthea data to solve some task.
Create a notebook which demonstrates the performance of this model.

Synthea data can be fairly large, so we should find the minimum number of patients we need to develop a useful model. We're just trying to demonstrate a concept, not cure a disease.

## Is your feature request related to a problem?
Before we attempt to develop a model on vertically-partitioned synthea data, we should confirm that a model can be trained successfully.

This model will also set the baseline so we can compare performance drop caused by privacy-preserving vertical federated learning

## What alternatives have you considered?
None

## Additional Context
An existing solution we could use is diabetes prediction https://github.com/IBM/example-health-machine-learning/blob/master/diabetes-prediction.ipynb

Requires  #58 to be complete

### PR DESCRIPTION
**Pull request by pavlos-p**
## Description
The example `Synthea_Medications_Tensorflow` notebook created that is trained on Synthea's `medications.csv`.
The notebook has guidelines to run it on Google Colab if needed. This utilises the GPU accelerator in order to train the model faster.

The Synthea download script has been modified to clone the Synthea repo (https://github.com/synthetichealth/synthea) and generate the datasets on `PyVertical/data` folder.

A new separated Tensorflow environment created (pyvertical-dev-tf) to demonstrate this notebook.

Some further example notebooks have been created in `PyVertical/examples/experimental` folder. The `Simple_Synthea_SKlearn` is a simple kNN classifier that is developed using Scikit-Learn. The `Synthea_Medications_PyTorch` demonstrates similar training as the Tensorflow notebook, but using PyTorch. It is currently broken, but it will be fixed in the future.

Closes #70 

## How has this been tested?
Tested in a fresh PyVertical fork and environment.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: scripts/download_synthea.sh [hunk 1]
```diff
@@ -1,14 +1,26 @@
 #!/bin/bash
+# Download Synthea Data in the correct directory PyVertical/data/synthea
+cd ../data/
+GITKEEP_exists=/synthea/.gitkeep
+if test -f "$GITKEEP_exists"; then
+    rm /synthea/.gitkeep
+fi
+
+git clone https://github.com/synthetichealth/synthea.git
```
Comment (TTitcombe) [inline scripts/download_synthea.sh:9]: I've currently got synthea added as a submodule, so we shouldn't need to clone it, should we?

Perhaps, to keep `PyVertical` clean, we should remove synthea as a submodule and instead just provide instructions on *what* script to run to generate the data and *where* to place the data?

So replace this script with instructions, and remove the synthea submodule. Do you think that's a good idea?

File: tf_environment.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+name: pyvertical-dev-tf
+channels:
+  - tensorflow
+dependencies:
+  - python=3.8
+  - numpy >=1.16
+  - pandas >= 1.0.0
+  - tensorflow
```
Comment (TTitcombe) [inline tf_environment.yml:8]: I would recommend installing `tensorflow` under pip within conda, because I don't think the conda binary is supported by the tensorflow team.

Also, we should pin tensorflow to either `>= 2.0.0` or `< 2.0.0` as `v2` is a significant API change

File: tf_environment.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: pyvertical-dev-tf
+channels:
+  - tensorflow
```
Comment (TTitcombe) [inline tf_environment.yml:3]: remove the `tensorflow` channel and add `conda-forge`, else conda won't be able to find numpy or pandas

File: tf_environment.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+dependencies:
+  - python=3.8
+  - numpy >=1.16
```
Comment (TTitcombe) [inline tf_environment.yml:6]: this environment should have `jupyterlab` so people can run the notebooks without doing anything other than installing this conda environment

**Review by TTitcombe — CHANGES_REQUESTED** [review]

**Comment by TTitcombe** [comment]
The test failure is nothing to do with this PR. Looks like another failing PSI test

File: tf_environment.yml (hunk 4 above)
Comment (pavlos-p) [reply tf_environment.yml:6]: Great, should we do the same for the PyTorch's environment.yml, instead of running `conda install notebook`?

File: scripts/download_synthea.sh (hunk 1 above)
Comment (pavlos-p) [reply scripts/download_synthea.sh:9]: Yes, makes sense. I have modified the script with the instructions instead of cloning synthea repo.

File: examples/experimental/Simple_Synthea_SKlearn.ipynb [hunk 5]
```diff
@@ -0,0 +1,402 @@
+{
```
[the remark this answers is not in the document]
Comment (pavlos-p) [reply examples/experimental/Simple_Synthea_SKlearn.ipynb:1]: Should we leave the `Synthea_Simple_Sklearn` on the `/experimental/` folder?

File: tf_environment.yml (hunk 2 above)
Comment (pavlos-p) [reply tf_environment.yml:8]: Fixed.

File: tf_environment.yml (hunk 3 above)
Comment (pavlos-p) [reply tf_environment.yml:3]: Fixed.

File: tf_environment.yml (hunk 4 above)
Comment (TTitcombe) [reply tf_environment.yml:6]: I forgot that I left it out for a reason - syft wasn't playing well with jupyterlab/notebook when conda was trying to resolve versions. Leave `jupyterlab` out for now

File: examples/experimental/Simple_Synthea_SKlearn.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (TTitcombe) [reply examples/experimental/Simple_Synthea_SKlearn.ipynb:1]: I think so. Experimental should be for non-VFL and VFL which isn't quite ready yet. Notebooks not placed in `experimental` are ones we're happy to show off as a good example of the PyVertical method
